Error while checking build/O2/o2 for 343b162c9a7a65a55f00f62cf903106735c7165a at 2022-10-13 09:26:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 457
100% tests passed, 0 tests failed out of 102


## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/10057-slc7_x86-64/0/PWGMM/Mult/DataModel/bestCollisionTable.h:50:32: error: redefinition of 'struct o2::aod::indices::CollisionIds'
/sw/SOURCES/O2Physics/10057-slc7_x86-64/0/PWGMM/Mult/DataModel/bestCollisionTable.h:50:32: error: redefinition of 'struct o2::aod::indices::CollisionIds'
/sw/SOURCES/O2Physics/10057-slc7_x86-64/0/PWGMM/Mult/DataModel/bestCollisionTable.h:50:32: error: redefinition of 'struct o2::aod::indices::CollisionIds'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10057/343b162c9a7a65a55f00f62cf903106735c7165a/build_O2_o2/pretty.html).

--- a/Framework/Core/include/Framework/AnalysisDataModel.h
+++ b/Framework/Core/include/Framework/AnalysisDataModel.h
@@ -1109,13 +1109,14 @@ using HepMCHeavyIon = HepMCHeavyIons::iterator;
 
 namespace indices
 {
-DECLARE_SOA_INDEX_COLUMN(Collision, collision); //!
-DECLARE_SOA_INDEX_COLUMN(BC, bc);               //!
-DECLARE_SOA_INDEX_COLUMN(Zdc, zdc);             //!
-DECLARE_SOA_INDEX_COLUMN(FV0A, fv0a);           //!
-DECLARE_SOA_INDEX_COLUMN(FV0C, fv0c);           //!
-DECLARE_SOA_INDEX_COLUMN(FT0, ft0);             //!
-DECLARE_SOA_INDEX_COLUMN(FDD, fdd);             //!
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);        //!
+DECLARE_SOA_ARRAY_INDEX_COLUMN(Collision, collisions); //!
+DECLARE_SOA_INDEX_COLUMN(BC, bc);                      //!
+DECLARE_SOA_INDEX_COLUMN(Zdc, zdc);                    //!
+DECLARE_SOA_INDEX_COLUMN(FV0A, fv0a);                  //!
+DECLARE_SOA_INDEX_COLUMN(FV0C, fv0c);                  //!
+DECLARE_SOA_INDEX_COLUMN(FT0, ft0);                    //!
+DECLARE_SOA_INDEX_COLUMN(FDD, fdd);                    //!
 } // namespace indices
 
 // First entry: Collision
@@ -1132,6 +1133,11 @@ DECLARE_SOA_INDEX_TABLE_EXCLUSIVE(MatchedBCCollisionsExclusive, BCs, "MA_BCCOL_E
                                   indices::BCId, indices::CollisionId);
 DECLARE_SOA_INDEX_TABLE(MatchedBCCollisionsSparse, BCs, "MA_BCCOL_SP", //!
                         indices::BCId, indices::CollisionId);
+
+DECLARE_SOA_INDEX_TABLE_EXCLUSIVE(MatchedBCCollisionsExclusiveMulti, BCs, "MA_BCCOLS_EX", //!
+                                  indices::BCId, indices::CollisionIds);
+DECLARE_SOA_INDEX_TABLE(MatchedBCCollisionsSparseMulti, BCs, "MA_BCCOLS_SP", //!
+                        indices::BCId, indices::CollisionIds);
 
 DECLARE_SOA_INDEX_TABLE_EXCLUSIVE(Run3MatchedToBCExclusive, BCs, "MA_RN3_BC_EX", //!
                                   indices::BCId, indices::ZdcId, indices::FT0Id, indices::FV0AId, indices::FDDId);

--- a/Framework/Core/include/Framework/IndexBuilderHelpers.h
+++ b/Framework/Core/include/Framework/IndexBuilderHelpers.h
@@ -70,6 +70,7 @@ struct SelfIndexColumnBuilder {
   }
 
   std::string mColumnName;
+  std::shared_ptr<arrow::DataType> mArrowType;
   std::unique_ptr<arrow::ArrayBuilder> mBuilder = nullptr;
 };
 
@@ -133,7 +134,6 @@ class IndexColumnBuilder : public SelfIndexColumnBuilder, public ChunkedArrayIte
   std::shared_ptr<arrow::ChunkedArray> resultMulti() const;
 
   int mListSize = 1;
-  std::shared_ptr<arrow::DataType> mArrowType;
   arrow::ArrayBuilder* mValueBuilder = nullptr;
   std::unique_ptr<arrow::ArrayBuilder> mListBuilder = nullptr;
 

--- a/Framework/Core/src/AODReaderHelpers.cxx
+++ b/Framework/Core/src/AODReaderHelpers.cxx
@@ -127,6 +127,10 @@ AlgorithmSpec AODReaderHelpers::indexBuilderCallback(std::vector<InputSpec>& req
           outputs.adopt(Output{origin, description, version}, maker(o2::aod::MatchedBCCollisionsExclusiveMetadata{}));
         } else if (description == header::DataDescription{"MA_BCCOL_SP"}) {
           outputs.adopt(Output{origin, description, version}, maker(o2::aod::MatchedBCCollisionsSparseMetadata{}));
+        } else if (description == header::DataDescription{"MA_BCCOLS_EX"}) {
+          outputs.adopt(Output{origin, description, version}, maker(o2::aod::MatchedBCCollisionsExclusiveMultiMetadata{}));
+        } else if (description == header::DataDescription{"MA_BCCOLS_SP"}) {
+          outputs.adopt(Output{origin, description, version}, maker(o2::aod::MatchedBCCollisionsSparseMultiMetadata{}));
         } else if (description == header::DataDescription{"MA_RN3_BC_SP"}) {
           outputs.adopt(Output{origin, description, version}, maker(o2::aod::Run3MatchedToBCSparseMetadata{}));
         } else if (description == header::DataDescription{"MA_RN3_BC_EX"}) {

--- a/Framework/Core/src/IndexBuilderHelpers.cxx
+++ b/Framework/Core/src/IndexBuilderHelpers.cxx
@@ -28,7 +28,8 @@ ChunkedArrayIterator::ChunkedArrayIterator(std::shared_ptr<arrow::ChunkedArray> 
 }
 
 SelfIndexColumnBuilder::SelfIndexColumnBuilder(const char* name, arrow::MemoryPool* pool)
-  : mColumnName{name}
+  : mColumnName{name},
+    mArrowType{arrow::int32()}
 {
   auto status = arrow::MakeBuilder(pool, arrow::int32(), &mBuilder);
   if (!status.ok()) {
@@ -38,7 +39,7 @@ SelfIndexColumnBuilder::SelfIndexColumnBuilder(const char* name, arrow::MemoryPo
 
 std::shared_ptr<arrow::Field> SelfIndexColumnBuilder::field() const
 {
-  return std::make_shared<arrow::Field>(mColumnName, arrow::int32());
+  return std::make_shared<arrow::Field>(mColumnName, mArrowType);
 }
 
 IndexColumnBuilder::IndexColumnBuilder(std::shared_ptr<arrow::ChunkedArray> source, const char* name, int listSize, arrow::MemoryPool* pool)
@@ -204,7 +205,11 @@ void IndexColumnBuilder::fillSlice(int idx)
 void IndexColumnBuilder::fillMulti(int idx)
 {
   (void)static_cast<arrow::ListBuilder*>(mListBuilder.get())->Append();
-  (void)static_cast<arrow::Int32Builder*>(mValueBuilder)->AppendValues(mIndices[idx].data(), mIndices[idx].size());
+  if (std::find(mValues.begin(), mValues.end(), idx) != mValues.end()) {
+    (void)static_cast<arrow::Int32Builder*>(mValueBuilder)->AppendValues(mIndices[idx].data(), mIndices[idx].size());
+  } else {
+    (void)static_cast<arrow::Int32Builder*>(mValueBuilder)->AppendValues(nullptr, 0);
+  }
 }
 
 std::shared_ptr<arrow::Int32Array> ChunkedArrayIterator::getCurrentArray()

